client/app/private-database/private-database.service.js

```diff
@@ -117,8 +117,12 @@ angular.module("services").service(
                         .then((_response) => {
                             database.serviceInfos = _response.data;
                             return database;
-                        })
-                        .catch((err) => this.$q.reject(err.data));
+                        });
+                })
+                .then((db) => {
+                    // reformat capabilities array as an object for easy access
```

Comment:
As we had a talk about comments yesterday, I feel like this is the good time to say that this kind of comment feels unnecessary

Reply:
I felt like I wanted to explain why I was adding a formatting step to the query, but I guess you're right, it's pretty obvious.